.github/workflows/main.yml

```diff
@@ -0,0 +1,15 @@
+on: [pull_request]
+
+jobs:
+  lint-and-tests:
+    name: 'Lint and Test'
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Build docker image
+        run: docker build . -t app-feup
+      - name: Lint
+        run: docker run -t app-feup flutter analyze --no-pub --preamble .
+      - name: Test
+        run: docker run -t app-feup flutter test
```

Comment:
Looks great! However, perhaps we can think of dropping our Dockerfile since there are already existing actions that do the same: https://github.com/marketplace/actions/flutter-action. I think it could interesting to go through this means so that we get macOS build support in the future. Wdyt?